### PR TITLE
XS-3135 DQC 33/36 needs a complicated test case.

### PR DESCRIPTION
**Pull request by mindeerickson-wf**
#### Changes:

- 1.  Changed the code in dqc_us_0033_0036.py to handle the case where a 36 can fire with dimensions and a 33 without dimensions can still fire as well
- 2.  Added a unit test to test the case of dimensions and having both a 36 and 33 fire

Please review: @hefischer  @andrewperkins-wf

### CODE REVIEW

File: dqc_us_rules/dqc_us_0033_0036.py [hunk 1]
```diff
@@ -131,8 +132,14 @@ def _doc_period_end_date_check(model_xbrl):
...
 
+                # If there are no dimensions on the DPED (DQC36) fact and the
+                # fact being checked for DQC33 has no dimensions, we do not
+                # want to fire DQC33 so we go on to the next fact
+                if no_dimensions and len(fact.context.segDimValues) == 0:
+                    continue
+
                 if check_for_lea_member(
-                    fact, not_valid_dped, fire_undimensionalized_33s
+                    fact, not_valid_dped
```
Comment (chrislococo-wf) [inline dqc_us_rules/dqc_us_0033_0036.py]: SG: You can compress this back into one line.

**Comment by chrislococo-wf** [comment]
+1

File: tests/unit_tests/test_dqc_us_0033_0036.py [hunk 2]
```diff
@@ -356,7 +466,7 @@ def test_check_for_lea_member(self):
         returned when the fact has the legal entity member
         """
         not_valid_dped = ['vault', 'bars', 'foo', 'beam', 'floor', 'boo']
-        fire_undimensionalized_33s = True
+        # fire_undimensionalized_33s = True
```
Comment (jasonleinbach-wf) [inline tests/unit_tests/test_dqc_us_0033_0036.py]: Commented code.

**Comment by bradbenjamin-wf** [comment]
I thought of another approach that might look cleaner.  If a DPED fact is found invalid in 36 what if we removed it from the dict so that the 33 loop just didn't even touch it?

I don't really like either of the names we've tried for that variable.  (fire_undimensionalized_33s or no_dimensions)  I also don't really care for check_for_lea_member in terms of contract.  Seems like the idea might be a way to remove both.

Don't let my comment hold back a +1 as I think the validation code is technically correct as is.  I haven't evaluated the unit test code.

**Comment by jasonleinbach-wf** [comment]
+1

**Comment by jasonleinbach-wf** [comment]
Minde is going to discuss that idea with Brad and we will follow up with it for next release.

**Comment by chrislococo-wf** [comment]
+1

**Comment by sagesmith-wf** [comment]
+10 Verified the test cases

**Comment by jasonleinbach-wf** [comment]
Reviewed and verified.  Merging.

**Pull request merged**